herbert_core/utilities/hdf_nexus/read_nexus_groups_recursive.m

```diff
@@ -0,0 +1,45 @@
+function datastruct = read_nexus_groups_recursive(filename, nexus_path)
+    % Recursively reads a NeXus file group and returns a structure of the data
+    %
+    % datastruct = read_nexus_group_recursive(hdf_info_struct)
+    % datastruct = read_nexus_group_recursive(filename, nexus_path)
+    % datastruct = read_nexus_group_recursive(filename, group_struct)
+    %
+    % Inputs:
+    %   hdf_info_struct - the result of h5info on the required group
+    %   filename (string) - the NeXus file name
+    %   nexus_path (string) - the NeXus path to the required group.
+    %   group_struct - the group structure of part of a h5info output
+    %
+    if nargin == 1 && isstruct(filename) && isfield(filename, 'Filename')
+        dinfo = filename;
+        filename = dinfo.Filename;
+    elseif isstring(filename) || ischar(filename)
+        if isstring(nexus_path) || ischar(nexus_path)
```

Comment:
No sure I understand... `nexus_path` must be a string/char if we use it with `h5info` (which produces a struct). Otherwise I'm assuming it's already a struct. If it's neither then there will be an error later on (in `read_nexus_datasets`).